### PR TITLE
Fix bugs with parsing to and from timezones

### PR DESCRIPTION
**Pull request by shepherdsam**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -432,125 +432,137 @@ Utils.parseDateTimeField = function(field, options) {
...
-  // Help SugarDate be aware of timezones
-  var previousNewDateInternal;
-  if (timezone) {
-    previousNewDateInternal = SugarDate.getOption('newDateInternal');
-    SugarDate.setOption('newDateInternal', function() {
-      return moment().tz(timezone).toDate();
-    });
+    var tz = moment.tz.guess() || 'UTC';
+    Utils.Logger.warn('Utils.parseDateTimeField - Timezone "' + timezone + '" is invalid. Assuming ' + tz);
+    timezone = tz;
```
Comment (fiznool) [inline lib/index.js]: We're changing `timezone` here but the `options.locale` was already decided above, with the old timezone. Does line 445 need to be moved to below this `if` block to take the new timezone into account?

File: lib/index.js [hunk 2]
```diff
@@ -432,125 +432,137 @@ Utils.parseDateTimeField = function(field, options) {
...
-  var dateMoment;
-  if (SugarDate.isValid(parsedDate)) {
+  var parsedDate = parseDateTime(field, options);
+  var rtnObject = getRtnObject(parsedDate, SugarDate.isValid(parsedDate));
+
+  var parsedDateWithTimezone = parseDateTime(field, options, timezone);
+  if (SugarDate.isValid(parsedDateWithTimezone)) {
+    Utils.Logger.debug('Utils.parseDateTimeField: parsedDateWithTimezone:', parsedDateWithTimezone.toISOString());
+
     var hasTZOffset = new RegExp('[+-]{1}[0-9]{2}:?[0-9]{2}').test(field);
```
Comment (fiznool) [inline lib/index.js]: I know this wasn't part of the changes but maybe this regex could be tightened up? How about:

```
/(?:[zZ]|[+-][01]\d:[0-5]\d)$/.test(field)
```

This takes into account:

- a UTC `Z` timezone offset
- the fact that `+99:99` isn't a valid timezone offset.

File: lib/index.js (hunk 1 above)
Comment (shepherdsam) [inline lib/index.js]: Yes, good catch!

File: lib/index.js (hunk 2 above)
Comment (shepherdsam) [inline lib/index.js]: Thanks, I ended up with.
```
/[zZ]|[+-][01]\d:?[0-5]\d/
```
I removed the `$` to allow for dates with our custom offset. `2013-02-08 09:30+07:00 +1d - 12h`.

**Pull request merged**